python/pyspark/sql/functions/builtin.py

```diff
@@ -18669,18 +18665,12 @@ def _invoke_higher_order_function(
 
     :return: a Column
     """
-    from py4j.java_gateway import JVMView
-    from pyspark.sql.classic.column import _to_java_column
+    from pyspark.sql.classic.column import _to_seq, _to_java_column
 
     sc = _get_active_spark_context()
-    expressions = cast(JVMView, sc._jvm).org.apache.spark.sql.catalyst.expressions
-    expr = getattr(expressions, name)
-
-    jcols = [_to_java_column(col).expr() for col in cols]
     jfuns = [_create_lambda(f) for f in funs]
-
-    return Column(cast(JVMView, sc._jvm).Column(expr(*jcols + jfuns)))
-
+    jcols = [_to_java_column(c) for c in cols]
+    return Column(sc._jvm.Column.pysparkFn(name, _to_seq(sc, jcols + jfuns)))
```

Comment:
@HyukjinKwon I cannot invoke `Column.fn(...)` here. Do you know why? I added `Column.pysparkFn(...)` as a workaround.

Reply:
Ah, the `Column.fn` has to be decorated with `scala.annotation.varargs` (which will create the same signature for variant arguments for Java).